### PR TITLE
Update Readme.md

### PR DESCRIPTION
**Pull request by anastasiakiseleva**
Updated the Readme.md file according to our guidelines.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-![](https://img.shields.io/endpoint?url=https://codecentral.devexpress.com/api/v1/VersionRange/128592237/22.2.4%2B)
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E622)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
 
-# XAF - How to prevent a Detail View from being shown for a List View record
+# XAF - Prevent Showing a Detail View of a List View Object
 
-## Scenario
+This example demonstrates how to prevent a user from invoking a Detail View of an object from a List View.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: "From" is an incorrect preposition in this scenario. When you speak of a user invoking a Detail View of an object -- you ask "Where?" but not "Where from"? Therefore, you answer the question "Where?" -- in a List View.

This logical flaw stems from Russian. Please take special care when you face such cases and start thinking in English. I remember that you facing this issue on regular basis. For example, you often said "to select an item **from** the menu", which is exactly the same error.

I ask you to track your own thoughts better. You cannot underestimate the importance of mental self-discipline. Thanks in advance.

```suggestion
This example demonstrates how to prevent a user from invoking a Detail View of an object in a List View.
```

File: Readme.md [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
 
-For certain data forms, a developer needs to limit end-users to editing only through the List View, i.e. without invoking a separate Detail View. This is usually done by activating the inline editing and `MasterDetailMode` = `ListViewAndDetailView` features as described in the following article: [List View Edit Modes](https://docs.devexpress.com/eXpressAppFramework/113249/ui-construction/views/list-view-edit-modes).
+To allow users to edit objects directly in a List View, enable the [in-place editing](https://docs.devexpress.com/eXpressAppFramework/113249/ui-construction/views/list-view-edit-modes#in-place-editing) and [split layout](https://docs.devexpress.com/eXpressAppFramework/113249/ui-construction/views/list-view-edit-modes#split-layout-the-masterdetailmode-property) functionality.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: I will not count this as your issue, however, this way it would have been better:

> the in-place edit and split layout functionality.

"in-place editing" as a standalone proper noun sounds fine, but when you use it as a compound adjective for "functionality", it sounds better in a non-gerund form -- "in-place edit functionality".

I hope this explanation makes sense.

```suggestion
To allow users to edit objects directly in a List View, enable the [in-place edit](https://docs.devexpress.com/eXpressAppFramework/113249/ui-construction/views/list-view-edit-modes#in-place-editing) and [split layout](https://docs.devexpress.com/eXpressAppFramework/113249/ui-construction/views/list-view-edit-modes#split-layout-the-masterdetailmode-property) functionality.
```

File: Readme.md [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
 
 ## Implementation Details
 
-To accomplish this task, access the [ListViewProcessCurrentObjectController class](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ListViewProcessCurrentObjectController) using the [Frame.GetController<ControllerType>() method](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Frame.GetController--1) and disable the ListViewProcessCurrentObjectController.ProcessCurrentObjectAction.
+Use the [Frame.GetController<ControllerType>](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Frame.GetController--1) method to access the [ListViewProcessCurrentObjectController](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ListViewProcessCurrentObjectController) class and disable the [ListViewProcessCurrentObjectController.ProcessCurrentObjectAction](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ListViewProcessCurrentObjectController.ProcessCurrentObjectAction) property.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: This sentence has a potential technical clarity issue -- you say that one should use the `Frame.GetController` method to access the mentioned class and disable some property...

But thing is, that you are most likely accessing an instance of the mentioned class rather than the class itself. The latter would have been correct if you rather changed a class property value, when a class is not a template for creating objects from it but sort of a namespace. From what I see here, this is not the case.

However, I could have been wrong because I don't know inner workings of your code in this scenario. Please clarify. Technical correctness and clarity are extremely important. Thanks in advance.

Also, I urge you to favor more accurate verbs over more generic verbs. In this particular scenario, would like to see "call" or "invoke" instead of "use" when it's related to a method.

File: Readme.md [hunk 4]
```diff
@@ -1,29 +1,27 @@
...
 
 ## Additional Information
+  
+This example only covers the case when an XAF application displays a Detail View of an object after a user double-clicks this object in a List View or focuses an object in a List View and presses Enter. To handle other scenarios, extend the controller's code. For example, if you do not want your XAF application to invoke a Detail View after a user clicks the **New** button to create an object in a List View, handle the [NewObjectViewController.ObjectCreating](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.NewObjectViewController.ObjectCreating) event and set its [ObjectCreatingEventArgs.ShowDetailView](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ObjectCreatingEventArgs.ShowDetailView) property to `false`.
```
Comment (dmitry-eliseev-devexpress) [inline Readme.md]: > This example only covers the case when an XAF application displays a Detail View of an object after a user double-clicks this object in a List View or focuses **an** object in a List View and presses Enter.

This is not an issue, but I can't help but explain that this scenario can be interpreted in two ways (and both would be completely correct):

* We started to talk about a specific object and its Detail View in this scenario, therefore, we speak of the same object throughout this sentence.
* We speak everywhere in general about an ability to do something (this is the scenario you explained).

I warn you that this article could be changed during the final editorial review, and there's nothing wrong with it.

> For example, if you do not want your XAF application to invoke a Detail View after a user clicks the **New** button to create an object in a List View, handle the [NewObjectViewController.ObjectCreating](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.NewObjectViewController.ObjectCreating) event and **set its** [ObjectCreatingEventArgs.ShowDetailView](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ObjectCreatingEventArgs.ShowDetailView) property to `false`.

But this looks a technical correctness issue to me because an event can't have a property per se. Therefore, you cannot set the event's property to anything. What do you actually do here, you set the named parameter accessible within the event handler to a value.

We can simplify this, of course, and rather say "set the X property to `false`.

```suggestion
This example only covers the scenario when an XAF application displays a Detail View of an object after a user double-clicks this object in a List View or focuses an object in a List View and presses Enter. To handle other scenarios, extend the controller's code. For example, if you do not want your XAF application to invoke a Detail View after a user clicks the **New** button to create an object in a List View, handle the [NewObjectViewController.ObjectCreating](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.NewObjectViewController.ObjectCreating) event and set the [ObjectCreatingEventArgs.ShowDetailView](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ObjectCreatingEventArgs.ShowDetailView) property to `false`.
```

**Review by dmitry-eliseev-devexpress — APPROVED** [review]
**Structure and Semantics**

This is a simple readme file for a code example. This topic has no structural issues and serves its purpose. However, I can point out a couple of issues related to technical correctness and terminology. Normally, you use a property to access a class instance rather than the class itself, save for a counted few exceptions. We need to clarify if this is the case. That's why I can't suggest an edit for it.

**Formatting**

We do not track formatting issues on GitHub, and this topic generally falls in line with countless similar topics we produce for examples.

**Language**

While I recognize your hard work on your language, and praise you for it, I can't help but notice a repeating mistake -- "from" instead of "in". This mistake follows you for a long time, therefore, I urge you to pay special attention to it. Please see my corresponding comment for details.

I can also note (with a certain degree of satisfaction) that I found no article-related issues. Well done!

**Bottom Line**

This is a good topic that I can approve right away. However, we need to clarify one technical question before moving further.

File: Readme.md (hunk 3 above)
Comment (anastasiakiseleva) [reply Readme.md]: This is indeed the case of accessing an instance of the class, Dmitry, you are correct, we should use more precise wording here.

```suggestion
Call the [Frame.GetController<ControllerType>](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Frame.GetController--1) method to access an instance of the [ListViewProcessCurrentObjectController](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ListViewProcessCurrentObjectController) class and disable the [ListViewProcessCurrentObjectController.ProcessCurrentObjectAction](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ListViewProcessCurrentObjectController.ProcessCurrentObjectAction) property.
```

Comment (dmitry-eliseev-devexpress) [reply Readme.md]: Yes, this change works, thank you. As a possible alternative, you could use the following pattern:

> to access a ClassName object.

**Review by dmitry-eliseev-devexpress — APPROVED** [review]

File: Readme.md [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
 
 ## Additional Information
+  
+This example only covers the scenario when an XAF application displays a Detail View of an object after a user double-clicks this object in a List View or focuses an object in a List View and presses Enter. To handle other scenarios, extend the controller's code. For example, if you do not want your XAF application to invoke a Detail View after a user clicks the **New** button to create an object in a List View, handle the [NewObjectViewController.ObjectCreating](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.NewObjectViewController.ObjectCreating) event and set the [ObjectCreatingEventArgs.ShowDetailView](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ObjectCreatingEventArgs.ShowDetailView) property to `false`.
```
Comment (arminal) [inline Readme.md]:
```suggestion
This example applies only when an XAF application displays a Detail View of an object after a user double-clicks this object in a List View, or focuses an object in a List View and presses Enter. To handle other scenarios, extend the controller's code. For example, if you do not want your XAF application to invoke a Detail View after a user clicks the **New** button to create an object in a List View, handle the [NewObjectViewController.ObjectCreating](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.NewObjectViewController.ObjectCreating) event and set the [ObjectCreatingEventArgs.ShowDetailView](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.SystemModule.ObjectCreatingEventArgs.ShowDetailView) property to `false`.
```

**Review by anastasiakiseleva — COMMENTED** [review]
readme.md updated
